server/plugin/api_test.go

```diff
@@ -716,3 +716,328 @@ func TestPlugin_handlePickerSelection(t *testing.T) {
 		})
 	}
 }
+
+func Test_handleDateTimeSelection(t *testing.T) {
+	defer monkey.UnpatchAll()
+
+	httpTestJSON := testutils.HTTPTest{
+		T:       t,
+		Encoder: testutils.EncodeJSON,
+	}
+
+	for name, test := range map[string]struct {
+		httpTest          testutils.HTTPTest
+		request           testutils.Request
+		expectedResponse  testutils.ExpectedResponse
+		userID            string
+		ParseAuthTokenErr error
+	}{
+		"User is unauthorized": {
+			httpTest: httpTestJSON,
+			request: testutils.Request{
+				Method: http.MethodPost,
+				URL:    fmt.Sprintf("/api/v1%s", PathDateTimeSelection),
+				Body: model.SubmitDialogRequest{
+					CallbackId: "mockPostID__Date",
+					Submission: map[string]interface{}{
+						"date": "2022-09-23",
+					},
+					ChannelId: "mockChannelID",
+				},
+			},
+			expectedResponse: testutils.ExpectedResponse{
+				StatusCode: http.StatusUnauthorized,
+			},
+		},
+		"Error parsing OAuth token": {
+			httpTest: httpTestJSON,
+			request: testutils.Request{
+				Method: http.MethodPost,
+				URL:    fmt.Sprintf("/api/v1%s", PathDateTimeSelection),
+				Body: model.SubmitDialogRequest{
+					CallbackId: "mockPostID__Date",
+					Submission: map[string]interface{}{
+						"date": "2022-09-23",
+					},
+					ChannelId: "mockChannelID",
+				},
+			},
```

Comment:
This is being repeated in each test case. Reduce this repitition.

Reply:
The date/time is changing in most cases. In case you missed that!

Reply:
Still, there's a lot of repitition. I can see same body in 3-4 test cases. Even if the date/time is changing, the rest of the body is same. You can reduce this repetition by using functions.

Reply:
Done